### DISCUSSION TITLE
Using type generics

### DISCUSSION THREAD

**Discussion by sctgraham** [supabase/supabase#5070] [General]
Hi all,

Has anyone been able to get type generics to work with the js client ? 
Here's a simplified version of what I'm trying do to :

```typescript
async function getThingsByStatus<T extends Record<string, any>(table: string, status: string[]) {
  const { data, error } = await supabase
    .from<T>(table)
    .select('*')
    .in('status', status)
  return data
}
```

The idea is then  to call `getThingsByStatus<MyTableType>('myTable', 'myStatus')` passing a type (`MyTableType`) which corresponds to the shape of the table being queried.

However the `.in()` method — but also all other filters — raises an error in typescript : 
`Argument of type 'string[]' is not assignable to parameter of type 'T[keyof T][]'`

Thanks for any help,

**Comment by brandy223** [comment]
Did you figured it out ?
